### ISSUE TITLE
Allow starting of container if foundryvtt.com is down, and release zip has already been downloaded

### ISSUE THREAD

**Issue by cs96and** [felddy/foundryvtt-docker#212]
# 🚀 Feature Proposal

foundryvtt.com went down for a few minutes earlier today, at the same time that I was trying to bring my container up.  The container failed to authenticate and gave up.

The release zip had already been downloaded from a previous run, and was in the cache, so it could have just carried on with the cached one.

**Comment by felddy** [comment]
This makes sense.  

I think the behavior you are looking for is the same as if `FOUNDRY_USERNAME` and/or `FOUNDRY_PASSWORD` are not specified.  As documented here: https://github.com/felddy/foundryvtt-docker#required-combinations

So if authentication fails all is not lost.  We can still fail forward to the cache.

**Comment by ThiefMaster** [comment]
Yes, failing forward to the cache would be great. But changing the lookup order to check the cache first would be even better:

I host 7 foundry foundry instances on my server. To avoid asking all my friends for their fvtt account login data, I use my own credentials for the download (but everyone uses their own license key of course).

When updating to a new foundry version, or even just restarting for some other reason (e.g. a server reboot / docker daemon restart), I frequently run into rate limiting (429 error). So pretty much the same problem like when the fvtt site is down.

While failing to the cache would help there, using the cache as the main source first (that's how caches work after all) would avoid spamming multiple logins and requests for a download url in the first place!

**Comment by cs96and** [comment]
Any chance of backporting this to the v12 container?  Thanks.

**Comment by felddy** [comment]
I don't have plans to back-port the feature to the `v12` container at this time.  But, I don't think there were any changes that would preclude the `v13` container from working with a `v12` release of FoundryVTT as is.  That means you can spin up a `v13` container and set the `FOUNDRY_VERSION` environment variable to `12.331`.  

This is similar to running a future version of FoundryVTT as mentioned in this discussion:
- https://github.com/felddy/foundryvtt-docker/discussions/796

As always, make sure to backup your data volume before testing this maneuver.   Also, be aware of the `v13` changes to the user, group, and permissions that are used by default: 
- https://github.com/felddy/foundryvtt-docker/discussions/1197

I'd be curious to know how it goes for you so we can document this for others that want to try.  

Good luck!

**Comment by cs96and** [comment]
Does the v13 container contain a version of Node that is compatible with v12?

**Comment by felddy** [comment]
The `v13` container uses Node `22`:

https://github.com/felddy/foundryvtt-docker/blob/32db205cb534ed5063bcc3bb3ca8845566677842/Dockerfile#L4

The Foundry team has said that FoundryVTT `v12` is compatible with Node `v22`:
- https://discord.com/channels/170995199584108546/689889940590690323/1354928111439057077
